extensions/common/gcp/gcp-core/src/test/java/org/eclipse/edc/gcp/common/GcpCredentialsTest.java

```diff
@@ -0,0 +1,134 @@
+                "\\nuOZQLeE+PyEvJegCRya+AVkCgYEA1R9WdxbUq5bEyiU8euIOe8jCfICADFVQ957+\\noZ4XMprXX/uNBa2IIEgkXW2uWYCL09fL+UlVcUu5iyVqXaRLBzD3/JW9IfCzCm1A\\nxaMpezv1vLyZp0n0VvTN2uag6t5M/FkYCmp+m+VzQi82dFm0DuTFulOix78lYVHA" +
+                "\\nwFKbwVcCgYByqdtczS+e02nN3M+XwrOnhnf/vwTj3BDtnXXF/MBp5SstHC1jDxLF\\nKGKUkcuCW9MKZkMo8Va5Fy6DeMp9IX9rrjye4f4QhSt5rEKDTjPZu9c4IObx5aBf\\nW6C1Ph/UfSWi50/w81+I2nuFUDikD4Y82qvkFfJp7foaw6jOVPTI2QKBgQC7kZQY" +
+                "\\nxbgwuEXEH1eGUxQqL3uz9ag8so3LEVzLQwbpm8t4Bz2LRLnsp3GR5KkwzmjB7kfv\\nw3H2f43x/+EIP0NlNdzbqbHGgEAjKhp6luo4MoJJNLgKupTYPyY5xQbVDwc0hPka\\nmbWKYTu6gTDs39IP1ZqMLXWzVPCCIWCCI3I/iwKBgArrjiqmWQ55xRbdGCvw5RCv" +
+                "\\ncIyuvwZTRYGrkGDIq7HHcFQDTqcxMMQhe0ox9WgQDuu8hAsrVcJv2Ia/14W3xQPT\\nyGAc9EtS112c3ZOYCyuPY4NgoIad9TvpHvpiaLNukbIUAIimoNbyWAUwmkTjTjkC\\n4X+WEhgfmekYzcrk6nfS\\n-----END PRIVATE KEY-----\\n\",\n" +
+                "  \"client_email\": \"test@project-test-1.iam.gserviceaccount.com\",\n" +
+                "  \"client_id\": \"client_id1\",\n" +
+                "  \"auth_uri\": \"https://accounts.google.com/o/oauth2/auth\",\n" +
+                "  \"token_uri\": \"https://oauth2.googleapis.com/token\",\n" +
+                "  \"auth_provider_x509_cert_url\": \"https://www.googleapis.com/oauth2/v1/certs\",\n" +
+                "  \"client_x509_cert_url\": \"https://www.googleapis.com/robot/v1/metadata/x509/project-test-1.iam.gserviceaccount.com\"\n" +
+                "}";
+
+        var tokenInJson = "{\"edctype\":\"dataspaceconnector:gcptoken\"" +
+                ",\"token\":\"" + tokenValue + "\"," +
+                "\"expiration\":1665998128000}";
+        serviceAccountFileInB64 = Base64.getEncoder().encodeToString(serviceAccountFileInJson.getBytes());
+        var tokenInB64 = Base64.getEncoder().encodeToString(tokenInJson.getBytes());
+
+
+        vault.storeSecret(accessTokenKeyName, tokenInJson);
+        vault.storeSecret(invalidAccessTokenKeyName, "{\"token\":\"" + tokenValue + "\"");
+        vault.storeSecret(serviceAccountKeyName, serviceAccountFileInJson);
+        vault.storeSecret(invalidServiceAccountKeyName, "{\"type\": \"service_account\" }");
+        gcpCredential = new GcpCredentials(vault, new TypeManager(), mock(Monitor.class));
+    }
+
+    @Test
+    void testResolveGoogleCredentialWhenTokenKeyNameIsProvided() {
+        var gcpCred = gcpCredential.resolveGoogleCredentialsFromDataAddress(new GcpServiceAccountCredentials(accessTokenKeyName, null, null));
+        assert (gcpCred != null);
+    }
+
+    @Test
+    void testResolveGoogleCredentialWhenInvalidTokenIsProvided() {
+        Exception thrown = assertThrows(EdcException.class, () -> gcpCredential.resolveGoogleCredentialsFromDataAddress(new GcpServiceAccountCredentials(invalidAccessTokenKeyName,  null, null)));
```

Comment:
extracting the `GcpServiceAccountCredentials` instance into a variable will make the test more readable